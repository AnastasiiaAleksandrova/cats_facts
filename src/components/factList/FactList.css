.facts {
    margin: 2em;
}

