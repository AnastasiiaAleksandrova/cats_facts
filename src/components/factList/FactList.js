import React from "react";
import Fact from "../fact/Fact";
import "./FactList.css";

function FactList(props) {

    return (
        <div className="facts">
            {props.facts.map(fact => (
                <Fact key={fact._id} text={fact.text} />
            ))}
        </div>
    );

}

export default FactList;
