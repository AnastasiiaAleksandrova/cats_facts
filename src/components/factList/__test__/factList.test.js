import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import FactList from "../FactList";
import renderer from 'react-test-renderer';

let container = null;

beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
});

afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
});

jest.mock("../../fact/Fact", () => {
    return function DummyFact(props) {
        return (
            <div key={props.key}>
                {props.text}
            </div>
        );
    };
});

it("renders list of facts", () => {
    const facts = [
        { __id: "1", text: "one" },
        { __id: "2", text: "two" },
        { __id: "3", text: "three" }
    ];
    render(<FactList facts={facts} />, container);
    expect(container.textContent).toContain("one", "two", "three");

});

it("matches snapshot", () => {
    const facts = [
        { __id: "1", text: "one" },
        { __id: "2", text: "two" },
        { __id: "3", text: "three" }
    ];
    const component = renderer.create(<FactList facts={facts} />).toJSON();
    expect(component).toMatchSnapshot();
})