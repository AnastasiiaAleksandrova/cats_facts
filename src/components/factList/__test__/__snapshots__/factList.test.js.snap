// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="facts"
>
  <div>
    one
  </div>
  <div>
    two
  </div>
  <div>
    three
  </div>
</div>
`;
